Sources/Service/WalletAvailabilityService.swift

```diff
@@ -3,8 +3,9 @@ import Networking
 import UIKit
 
 protocol FeatureAvailabilityService {
+    var hasAccessedBefore: Bool { get }
     var shouldShowFeature: Bool { get }
-    func hasAccessedPreviously()
+    func accessedWalletFeature()
```

Comment:
Considering this is a protocol for a "feature" availability service I'm not sure if we should use the word "wallet" for conforming types as they could be features other than wallet.
The "wallet" specificity would come from the type which conforms to this - in the example below it's `WalletAvailabilityService` so it's described at the call sight already 🙂 
```suggestion
    func accessedFeature()
```

Reply:
Done, thank you 🙂